### PR TITLE
Fixed the way images are saved, adding the correct extension type at the end of the fileName

### PR DESCRIPTION
**Pull request by wprod**
## Fixed the way images are saved, adding the correct extension type at the end of the fileName

Images are now saved with the right extension type, who is extracted from the MIME.

### CODE REVIEW

File: src/Services/UploadImageService.php [hunk 1]
```diff
@@ -208,6 +217,9 @@ protected function setUploadedImage(string $imageAttributeName)
             $originalValue = $this->model->getOriginal($imageAttributeName);
         }
 
+        $path_parts = pathinfo($value);
+        $path_extension = array_get($path_parts, 'extension');
```
Comment (jguyomard) [inline src/Services/UploadImageService.php]:
```
$pathExtension = pathinfo($value, PATHINFO_EXTENSION);
```

File: src/Services/UploadImageService.php [hunk 2]
```diff
@@ -223,14 +235,14 @@ protected function setUploadedImage(string $imageAttributeName)
         }
 
         // An image is already uploaded, a new one is uploaded
-        if (ends_with($value, '.jpg') && !empty($originalValue)) {
+        if (str_contains($path_extension, $this->allowed_extensions) && !empty($originalValue)) {
```
Comment (jguyomard) [inline src/Services/UploadImageService.php]: Why `str_contains` ? and not `in_array`?

File: src/Services/UploadImageService.php (hunk 1 above)
Comment (wprod) [reply src/Services/UploadImageService.php]: Thanks :)

File: src/Services/UploadImageService.php (hunk 2 above)
Comment (wprod) [reply src/Services/UploadImageService.php]: $path_extension can have a "?v=5665..." parameter. So it'll fail if we use in_array. Plus, str_contains is a cool laravel helper :sunglasses:

Comment (jguyomard) [reply src/Services/UploadImageService.php]: In this case, prefers `starts_with` (because current code handle `toto.exe?v=jpg`)

Comment (wprod) [reply src/Services/UploadImageService.php]: Thanks :)

File: src/Services/UploadImageService.php [hunk 3]
```diff
@@ -11,6 +11,13 @@
 class UploadImageService extends AbstractUploadService
 {
     /**
+     * Allowed image extensions
+     *
+     * @var array
+     */
+    protected $allowed_extensions = ['jpeg', 'jpg', 'gif', 'png'];
```
Comment (tonyyb) [inline src/Services/UploadImageService.php]: CamelCase here.

**Review by tonyyb — CHANGES_REQUESTED** [review]

**Review by cholletNovius — APPROVED** [review]

**Review by tonyyb — APPROVED** [review]

**Pull request merged**